packages/next/src/client/components/navigation.ts

```diff
@@ -158,9 +152,17 @@ function getSelectedParams(
   return getSelectedParams(node, params)
 }
 
-export function useParams() {
+/**
+ * Get the current parameters. For example useParams() on /dashboard/[team]
+ * where pathname is /dashboard/nextjs would return { team: 'nextjs' }
+ */
+export function useParams(): Params {
   clientHookInServerComponentError('useParams')
   const { tree } = useContext(GlobalLayoutRouterContext)
+  if (!tree) {
+    // This only happens in `pages`. Type is overwritten in navigation.d.ts
+    return null!
```

Comment:
I think so, but @timneutkens pointed out that it will include search params because the old router doesn't differentiate. But users can implement this themselves with `next/compat/router` if they need to.

Reply:
The above would only include the dynamic route params from the query not all `searchParams`